### PR TITLE
Restructure contribution guide

### LINKED ISSUE #40: rework developer documentation to prioritize common contributions (docs, webapp)
## Description

Our developer docs are in need of some attention mostly because we've made gradual changes over the past couple of years and as such the overall flow is confusing. Specifically, we should rewrite/reorganize them to more clearly distinguish the following scenarios:

1. Contributor wants to edit docs only.
2. Contributor wants to work on web app changes only (in which case they can just use the docker dev env and don't need to e.g. set up vagrant, rsync, etc. as described here: https://docs.securedrop.org/en/latest/development/setup_development.html#id2) - this is the most common development that happens in hackathons and drive-by contributions. 
3. Contributor wants to work on system configuration / provisioning - in this case they do need to be able to spin up staging. It is very rare for a hackathon participant to do contributions in this area due to the overhead involved. 

Since 1 and 2 are where most contributions happen, we should try to make those areas of the docs as clear as possible to someone unfamiliar with the project.

### PR DESCRIPTION
**Pull request by joaedwar**
## Status

Ready for review

## Description of Changes

* Description: Restructures the introductory module for the contributor guide

Partially fixes the following:
* Fixes #40

## Testing

*

## Release 

* 

## Checklist (Optional)

- [x] Doc linting (`make docs-lint`) passed locally
- [ ] Doc link linting (`make docs-linkcheck`) passed
- [x] You have previewed (`make docs`) docs at http://localhost:8000

### CODE REVIEW

**Comment by joaedwar** [comment]
@eloquence

**Comment by eloquence** [comment]
Hi @joaedwar and happy new year! Are you still interested in working on this one?

**Comment by joaedwar** [comment]
@eloquence Happy new year! Yes, I am.  I added some changes right before the break; I'll rebase again. Please let me know if you have further suggestions for the content.

**Comment by eloquence** [comment]
Awesome, I'll take another spin once it's rebased and linter is happy, let me know if I can help with that. :)

File: docs/development/contributing.rst [hunk 1]
```diff
@@ -75,35 +97,42 @@ If you would like to contribute on a regular basis, you'll want to read the
...
 Technical writers and editors are invited to review the `documentation
 <https://docs.securedrop.org/>`__ and fix any mistakes in accordance with the
-:doc:`documentation guidelines <documentation_guidelines>`.
+:doc:`documentation guidelines <documentation_guidelines>`. Our documentation code is 
+located in our `documentation repository <https://github.com/freedomofpress/securedrop-docs>`__.
 
-If this is your first time helping with SecureDrop documentation, consider
-working on `low-hanging fruit`_ to become familiar with the process.
+If this is your first time contributing to SecureDrop documentation, consider
+working on `low-hanging fruit`_ to become familiar with the process. Refer to our
```
Comment (gonzalo-bulnes) [inline docs/development/contributing.rst:110]:
```suggestion
working on a `low-hanging fruit`_ to become familiar with the process. Refer to our
```

Again, not 100% sure if this suggestion should be applied, but it seems to me that "a _low-hanging fruit_" (or "_low-hanging fruits_") fit better in the sentence?

**Review by gonzalo-bulnes — COMMENTED** [review]
👋 I'm adding a few comments to highlight an extra/missing word in a couple of places.

File: docs/development/contributing.rst (hunk 1 above)
Comment (eloquence) [reply docs/development/contributing.rst:113]: @gonzalo-bulnes "Fruit" is [usually uncountable](https://www.ldoceonline.com/dictionary/fruit), so in English the sentence as written works better than your suggested change. This PR currently breaks that link by removing the link target, however.

**Comment by eloquence** [comment]
Thanks @joaedwar for your work on this and @gonzalo-bulnes for your initial comments.

I've cleaned this up a bit and it is now ready for a final pass:
- Fixed broken link
- Fixed formatting issues
- Added reference to Code of Conduct early on
- Emphasized communication more strongly as suggested in https://github.com/freedomofpress/securedrop/issues/5681
- Tweaked some headings
- Rebased and squashed into two commits

If either of you have time for peer reviewing the changes in ab466031b6eb88aaa9d5afa50e348da43929026a, that would be awesome :)

**Comment by joaedwar** [comment]
lgtm

File: docs/development/contributing.rst [hunk 2]
```diff
@@ -34,37 +42,60 @@ SecureDrop's code is written in `Python`_.
...
+Preparing and submitting changes
+--------------------------------
+Before beginning your work on any given issue, we recommend asking questions
+or sharing an implementation proposal on the relevant GitHub issue.
+Alternatively, you can often find the development team on `Gitter chat <https://gitter.im/freedomofpress/securedrop>`__.
+Communicating early and often is especially important for larger changes.
+
 When you're ready to share your work with the SecureDrop team for review, submit
 a `pull request
 <https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests>`__
```
Comment (gonzalo-bulnes) [inline docs/development/contributing.rst:101]: GitHub has set up a permanent redirection (`make docs-linkcheck`).

```suggestion
<https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests>`__
```

**Review by gonzalo-bulnes — APPROVED** [review]
Yes, looks good to me as well :+1: 

- the formatting looks good and consistent
- the overall page makes sense when read from to to bottom
- `make docs-linkcheck` is happy - a few permanent redirects sneaked in, but only one in this page. (I'd fix it because permanent redirects are quite noisy in the linkcheck output.)

**Review by eloquence — APPROVED** [review]
Thanks @joaedwar for the changes and thanks @gonzalo-bulnes for the peer review :)

**Pull request merged**